### ISSUE TITLE
Overflow error on distributed choice 

### ISSUE THREAD

**Issue by joey-coleman** [symphonytool/symphony#170]
> The following bug was [originally reported](https://sourceforge.net/p/compassresearch/bugs/170/) on Sourceforge by [*anonymous](https://sourceforge.net/u/*anonymous/profile/), 2013-11-26 18:16:44.952000:

channels
 on : nat
The interpreter can't manage distributed choice.  The process below causes a StackOverflowError.

process Spec = 
 begin
 state  

 all_nodes_nat : set of nat := {0,1,2}

 actions

```
NEWAlive = alive : set of nat @ 
  ([] i in set (alive) @ on!i -> NEWAlive(alive))

NEWAliveTwo = alive : set of nat @ 
  (on!1 -> NEWAliveTwo(alive union {1}))
   []
   (on!2 -> NEWAliveTwo(alive))
  [] 
  (on!0 -> NEWAliveTwo(alive))
```

  @ 
   NEWAlive(all_nodes_nat)  -- begin with no nodes on and no time passed  
 end

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/170/#81b3) by [anderskm](https://sourceforge.net/u/anderskm/profile/), 2013-11-27 13:08:59.818000:
- **status**: open --> closed
- **assigned_to**: Anders Kaels Malmos

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/170/#7130) by [anderskm](https://sourceforge.net/u/anderskm/profile/), 2013-11-27 13:09:00.190000:

fixed on [85c04a23187aaf2b439aee8bbbec2c16021ca572]
